File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/server/consumes/ConsumesController.kt
package io.micronaut.docs.server.consumes

//tag::imports[]
import io.micronaut.context.annotation.Requires
import io.micronaut.http.HttpResponse
import io.micronaut.http.MediaType
import io.micronaut.http.annotation.Consumes
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Post
//end::imports[]

@Requires(property = "spec.name", value = "consumesspec")
//tag::clazz[]
@Controller("/consumes")
class ConsumesController {

    @Post // <1>
    fun index(): HttpResponse<*> {
        return HttpResponse.ok<Any>()
    }

    @Consumes(MediaType.APPLICATION_FORM_URLENCODED, MediaType.APPLICATION_JSON) // <2>
    @Post("/multiple")
    fun multipleConsumes(): HttpResponse<*> {
        return HttpResponse.ok<Any>()
    }

    @Post(value = "/member", consumes = [MediaType.TEXT_PLAIN]) // <3>
    fun consumesMember(): HttpResponse<*> {
        return HttpResponse.ok<Any>()
    }
}
//end::clazz[]

File: test-suite-python/src/test/python/micronaut/docs/server/consumes/ConsumesController.py
# tag::imports[]
from micronaut.context.annotation import Requires
from micronaut.http import HttpResponse, MediaType
from micronaut.http.annotation import Consumes, Controller, Post
# end::imports[]


@Requires(property="spec.name", value="consumesspec")
# tag::clazz[]
@Controller("/consumes")
class ConsumesController:

    @Post  # <1>
    def index(self) -> HttpResponse:
        return HttpResponse.ok()

    @Consumes([MediaType.APPLICATION_FORM_URLENCODED, MediaType.APPLICATION_JSON])  # <2>
    @Post("/multiple")
    def multipleConsumes(self) -> HttpResponse:
        return HttpResponse.ok()

    @Post(value="/member", consumes=MediaType.TEXT_PLAIN)  # <3>
    def consumesMember(self) -> HttpResponse:
        return HttpResponse.ok()
# end::clazz[]
